core/src/scene/sceneLoader.cpp

```diff
@@ -200,12 +208,15 @@ MaterialTexture SceneLoader::loadMaterialTexture(YAML::Node matCompNode, Scene&
 
     std::string name = textureNode.as<std::string>();
 
-    auto& sharedTexs = scene.getTextures();
-    if (sharedTexs.find(name) == sharedTexs.end()) {
-        sharedTexs.emplace(name, std::make_shared<Texture>(name));
+    auto& tex = scene.textures()[name];
+
+    if (!tex) {
+        tex = std::make_shared<Texture>(name);
     }
 
-    matTex.tex = sharedTexs.at(name);
+    matTex.tex = tex;
+
+    if (!matTex.tex) { matTex.tex.reset(new Texture(name)); }
```

Comment:
I am not sure I get this :(.
Isn't this same as [Line: 214](https://github.com/tangrams/tangram-es/pull/237/files#diff-7f29a0cf96bc2f3ea18f48b14ef40906R214)?

214: makes a shared ptr of a texture (if it does not exist already).
Line 219 also does the same. Shouldn't Line:217 suffice?